tests/test_stats.py

```diff
@@ -68,6 +71,8 @@ def test_inputs(ysize, xsize):
     X = np.random.randn(N, xsize)
 
     ccov = conditional_cov(X, Y)
+    if (isinstance(ccov, pd.DataFrame)):
+        ccov = ccov.to_numpy()
```

Comment:
See my previous comment.

Also, what may be worth doing, either in this function or later, is to explicitly test that if Y is a dataframe, that return type is a dataframe.

You could do something like:
```
Ydf = pd.DataFrame(Y, columns=["y", "z"])
... run the functions and test for the right output and column/index names
```

(I wouldn't bother re-testing the correct values are returned)